plugins/presentation/src/components/internal/Table/Table.svelte

```diff
@@ -0,0 +1,201 @@
+
+  const dispatch = createEventDispatcher()
+
+  export let attributes: AttrModel[] = []
+  export let rows: any[] = []
+  export let _class: Ref<Class<Doc>>
+  export let editable = false
+
+  export let page = 0
+  // the shown page index
+  export let pageIndex = 0
+  // rows amount at the page
+  export const pageSize = 10
+  export let labels = {
+    empty: 'No records available'
+  }
+
+  // all rows after searching
+  $: filteredRows = rows
+  // filtered rows at the page
+  $: visibleRows = filteredRows.slice(pageIndex, pageIndex + pageSize)
+
+  setContext('table-state', {
+    getState: () => ({
+      page,
+      pageIndex,
+      pageSize,
+      rows,
+      filteredRows
+    }),
+    setPage: (_page, _pageIndex) => {
+      page = _page
+      pageIndex = _pageIndex
+    },
+    setRows: (_rows) => (filteredRows = _rows)
+  })
+
+  function onPageChange (event) {
+    dispatch('pageChange', event.detail)
+  }
+
+  function onSearch (event) {
+    // console.log(event)
+    dispatch('search', event.detail)
+  }
+
+  $: console.log(visibleRows)
+
+  const handleRowClick = (_id: any) => () => {
+    dispatch('open', { _id, _class })
+  }
+
+  function onSort (event) {
+    event.detail.rows = sortString(event.detail.rows, event.detail.dir, event.detail.key)
+  }
+</script>
+
+<div class="table-component">
+  <!-- control panel-->
+  <slot name="controls">
+    <svelte:component this={TableControls} on:search={onSearch} />
+  </slot>
+
+  <table>
+    <thead>
+      <tr>
+        {#each attributes as attr (attr.key)}
+          <th>
+            {attr.label}
+            <Sort key={attr.label} on:sort={onSort} />
+          </th>
+        {/each}
+      </tr>
+    </thead>
+    {#if visibleRows.length === 0}
+      <tbody>
+        <tr>
+          <td class="center" colspan="100%">
+            <span>
+              {@html labels.empty}
+            </span>
+          </td>
+        </tr>
+      </tbody>
+    {:else}
+      <tbody>
+        {#each visibleRows as object (object._id)}
+          <tr on:click={handleRowClick(object._id)}>
+            {#each attributes as attr (attr.key)}
+              <td>
+                {#if attr.presenter}
+                  <Presenter is={attr.presenter} value={object[attr.key]} attribute={attr} {editable} />
+                {:else}
+                  <span>{object[attr.key] || ''}</span>
+                {/if}
+              </td>
+            {/each}
+          </tr>
+        {/each}
+      </tbody>
+    {/if}
+  </table>
+
+  <!--pagination-->
+  <slot name="bottom">
+    <div class="slot-bottom">
+      <svelte:component
+        this={Pagination}
+        {page}
+        {pageSize}
+        count={filteredRows.length - 1}
+        on:pageChange={onPageChange} />
+    </div>
+  </slot>
+</div>
+
+<style lang="scss">
+  .table-component {
+    padding: 20px;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    min-width: 1180px;
```

Comment:
Why is this minimum width required? I think, in general, the table should not have a minimum size, but I think it would be good to be able to set the minimum width for specific columns